deployment/terraform/data.tf

```diff
@@ -0,0 +1,13 @@
+data "terraform_remote_state" "core" {
+  backend = "s3"
+
+  config {
+    region = "${var.aws_region}"
+    bucket = "${var.remote_state_bucket}"
+    key    = "terraform/core/state"
+  }
+}
+
+data "aws_route53_zone" "external" {
+  zone_id = "${data.terraform_remote_state.core.public_hosted_zone_id}"
```

Comment:
If it simplifies things, doing this lookup via `name` vs. `zone_id` is fair game. It is a little tricky to use `name` for private hosted zones because technically there can be more than one active at a time (associated with different VPCs).